### PR TITLE
feat(sdk): Gas estimate tutorial

### PR DESCRIPTION
**Pull request by qbzzt**
Replacement for https://github.com/ethereum-optimism/optimism-tutorial/pull/89

Closing: INT-548

Commit history:

1. [Initial version](https://github.com/ethereum-optimism/optimism-tutorial/pull/89/commits/30ab5f15234d32567b114f275ec8eb9cb238754d)
1. [feat(sdk): Specified version 0.2.5](https://github.com/ethereum-optimism/optimism-tutorial/pull/89/commits/d28daef225aad7f27808feb4a0e7d168f67c213f)
1. [feat(sdk): Update to SDK 1.0.0](https://github.com/ethereum-optimism/optimism-tutorial/pull/89/commits/c3c86225062b6645df2946cfd2814807726bd99a)
1. feat(sdk): Added links to sdk.optimism.io

**Description**
A clear and concise description of the features you're adding in this pull request.

**Additional context**
Add any other context about the problem you're solving.

**Metadata**
- Fixes #[Link to Issue]

### CODE REVIEW

**Comment by qbzzt** [comment]
When this is merged: 

[src/docs/developers/bridge/messaging.md](https://github.com/ethereum-optimism/community-hub/pull/353/files/f12946a31a0cbc6be0d815d7958b5ba9e6169424#diff-d84f69f251345cde76779748502e1ef70631022b26da77f3ae5456b5064a9d43)
The first 1.92 million gas on L2 is free.
The vast majority of L1 to L2 calls spend less than that on L2, so nothing further is required.

If you think that your call might spend more than that on L2, you can specify a higher gas limit.
 
Member
@[Nicca42](https://github.com/Nicca42) Nicca42 [9 days ago](https://github.com/ethereum-optimism/community-hub/pull/353#discussion_r822090850)
Can we add a sentence like "If you'd like to check how much gas your transaction will use please see our JavaScript SDK gas estimator" Or whatever is most accurate + a link

 
Member
Author
@[qbzzt](https://github.com/qbzzt) qbzzt [2 days ago](https://github.com/ethereum-optimism/community-hub/pull/353#discussion_r826545420)
Yes, but let's do this after that tutorial (https://github.com/ethereum-optimism/optimism-tutorial/pull/89) is merged. I'm going to leave this unresolved to remind me to do that once it's merged.

File: sdk-estimate-gas/README.md
Comment (smartcontracts) [inline sdk-estimate-gas/README.md:20]: Can alternatively link to the faucets page ("... please replenish it from once of the faucets found here"). We will have more testnet faucets soon.

File: sdk-estimate-gas/README.md
Comment (smartcontracts) [inline sdk-estimate-gas/README.md:74]:
```suggestion
  This is typical in Optimistic transactions, because of the cost ratio between L1 gas and L2 gas.
```

File: sdk-estimate-gas/gas.js
Comment (smartcontracts) [inline sdk-estimate-gas/gas.js:39]: Slightly cleaner to simply return:

```
return {
  totalCost: ...
  l1Cost: ...
}
```
etc.

**Comment by Nicca42** [comment]
@smartcontracts Good to go?

**Review by smartcontracts — APPROVED** [review]

**Review by Nicca42 — APPROVED** [review]

**Pull request merged**
